Buildings/Controls/OBC/CDL/Integers/Sources/TimeTable.mo

```diff
@@ -0,0 +1,207 @@
+within Buildings.Controls.OBC.CDL.Integers.Sources;
+block TimeTable "Table look-up with respect to time with constant segments"
+
+  parameter Real table[:,:]
+    "Table matrix with time as a first table column (in seconds, unless timeScale is not 1) and Integers in all other columns";
+  parameter CDL.Types.Extrapolation extrapolation=CDL.Types.Extrapolation.HoldLastPoint
+    "Extrapolation of data outside the definition range";
+  parameter Integer offset[:]=fill(0, nout) "Offsets of output signals";
+  parameter Modelica.SIunits.Time timeScale=1
+    "Time scale of first table column. Set to 3600 if time in table is in hours";
+
+  Interfaces.IntegerOutput y[nout] "Output of the table"
+    annotation (Placement(transformation(extent={{100,-20},{140,20}})));
+
+protected
+  final parameter Integer nout=size(table, 2)-1
+    "Dimension of output vector";
+  final parameter Integer n=size(table, 1)
+    "Number of table points";
+  parameter Modelica.SIunits.Time t0(fixed=false)
+    "First sample time instant";
+
+  Continuous.Abs abs1[nout]
+    "Absolute values"
+    annotation (Placement(transformation(extent={{-80,-80},{-60,-60}})));
+  Continuous.Add add2[nout](
+    final k1=fill(1, nout),
+    final k2=fill(-1, nout))
+    "Subtracts inputs from their integer conversion results"
+    annotation (Placement(transformation(extent={{40,-40},{60,-20}})));
+  Conversions.IntegerToReal intToRea[nout] "Type conversion"
+    annotation (Placement(transformation(extent={{0,20},{20,40}})));
+  Continuous.GreaterThreshold greThr[nout](
+    final t=fill(Constants.small, nout))
+    "Value comparisson"
+    annotation (Placement(transformation(extent={{-40,-80},{-20,-60}})));
+  Utilities.Assert assMes(message="Scheduled values are not all of type Integer")
+    "Assert all scheduled values are integer values"
+    annotation (Placement(transformation(extent={{60,-80},{80,-60}})));
+  Logical.MultiOr mulOr(
+    final nu=nout) "Multi or"
+    annotation (Placement(transformation(extent={{0,-80},{20,-60}})));
+  Logical.Not not1 "Logical not"
+    annotation (Placement(transformation(extent={{30,-80},{50,-60}})));
```

Comment:
Simply make a loop in the initial equation section, this will be more efficient.